### ISSUE TITLE
Question about loss and activation function

### ISSUE THREAD

**Issue by naoki7090624** [DQiaole/ZITS_inpainting#12]
Hi,
I have questions about activation function and loss. 
1. Why you calculate the loss before the activation function?
According to your code, the cross entropy loss is calculated before sigmoid function. In the general CNN, I think the loss is calculated after the activation. Could you tell me why.

2. Why you use only cross entropy loss?
According to your code, only the cross entropy loss is calculated in the TSR. I wonder if you could use other losses (L1 loss, feature matching loss) after upsampling because there are convolution layers after transformer blocks.

**Comment by ewrfcas** [comment]
1. Hi, the loss function of TSR is 'binary_cross_entropy_with_logits', which uses logits (before the sigmoid) as the input.
If you want to use probs after the sigmoid (activation), we recommend using 'binary_cross_entropy' instead of 'binary_cross_entropy_with_logits'.

2. Feature matching loss should be used with a discriminator of adversarial training, but our TSR training is not an adversarial training, and no discriminator is needed here. For the L1 loss, we think that the BCE is more robust and easy to be controlled for the post-processing (e.g., multiply specified values before the sigmoid for the binarization). We also do some ablations and BCE loss can achieve slightly better results compared with L1 loss.

**Comment by naoki7090624** [comment]
Thanks a lot.
Is there any chance of improving accuracy by adding the L1 loss (or another loss) to the BCE loss as follows?
```
BCE_loss = F.binary_cross_entropy_with_logits(x,y)
L1_loss = nn.L1Loss(x,y)
loss = BCE_loss + L1_loss # + other losses
loss.backward()
```

**Comment by ewrfcas** [comment]
Both BCE and L1 loss work for the same target, so I think the improvement is limited.
Besides, using regression loss (L1) and classification loss (BCE) at the same time may cause some problems in adjusting loss weights between them.

**Comment by naoki7090624** [comment]
Thank you so much for clear and informative answers!!
